### ISSUE TITLE
read old tilia (.til) files?

### ISSUE THREAD

**Issue by orb16** [ropensci/neotoma#219]
A feature request for your consideration: a function to read old tilia files? Old as in version 1.0, '.til' rather than '.tilx'. I have a dosbox to run DOS tilia, but it would be awesome to loop through (all) the files and read them in. I've tried the `read.tilia` function from this package (fails) and `read.Tilia` from the`rioja`. The `read.Tilia` doesn't fail but doesn't bring in all the data.

**Comment by SimonGoring** [comment]
If you could attach a version of a file you would like read I'll see what I can do.  I don't have a lot of bandwidth right now though :/  I can look at `read.Tilia()` as well.

**Comment by orb16** [comment]
Yes of course! I need to check with my boss which are ok to put on here - will be Monday when I know. Cheers.

**Comment by orb16** [comment]
Please see the attached zip folder (includes data and an R script which demonstrates the problem [github_tilia_issue.zip](https://github.com/ropensci/neotoma/files/888790/github_tilia_issue.zip)). There's two problems with `read.Tilia()`: 

(1) Illegal characters are returned differently in the returned "data" element (using the nomenclature from the R help entry) compared to the "vars" argument. This is problematic as no merges can be done - any illegal names won't match. Moreso when extra rows have been added to either the data or the vars in Tilia originally, meaning even if it is safe to `cbind()` the columns, it is no longer possible. An `inner_join()` or similar would get around this, but for the illegal characters

(2) Some files are so corrupt they cannot be opened (so corrupt they won't open DOSTILIA or C2). That's ok - they are superfluous, but the C++ code appears to leave connections open if when a file fails, meaning if I am looping through a list of .TIL files some of which are corrupt, eventually the loop fails with a "too many open connections" error. I have tried closing all the connections but that doesn't seem to work.

I have also emailed Steve Juggins who is the package maintainer of `rioja`, and let him know the files and code are here: ideally at least one package might have the capacity to read old .til files? As noted (example in the script) `read.tilia()` just fails with an error.

Background: I'm working to 'revive' a bunch of .til files so they are available for use in the future.
